libz-rs-sys/src/gz.rs

```diff
@@ -1861,6 +1863,186 @@ pub unsafe extern "C-unwind" fn gzputs(file: gzFile, s: *const c_char) -> c_int
     }
 }
 
+/// Read one decompressed byte from `file`.
+///
+/// Note: The C header file zlib.h provides a macro wrapper for gzgetc that implements
+/// the fast path inline and calls this function for the slow path.
+///
+/// # Returns
+///
+/// - The byte read, on success.
+/// - `-1` on error.
+///
+/// # Safety
+///
+/// - `file`, if non-null, must be an open file handle obtained from [`gzopen`] or [`gzdopen`].
+#[cfg_attr(feature = "export-symbols", export_name = crate::prefix!(gzgetc))]
+pub unsafe extern "C-unwind" fn gzgetc(file: gzFile) -> c_int {
+    let Some(state) = (unsafe { file.cast::<GzState>().as_mut() }) else {
+        return -1;
+    };
+
+    // Check that we're reading and that there's no (serious) error.
+    if state.mode != GzMode::GZ_READ || (state.err != Z_OK && state.err != Z_BUF_ERROR) {
+        return -1;
+    }
+
+    // Try output buffer (no need to check for skip request).
+    if state.have != 0 {
+        state.have -= 1;
+        state.pos += 1;
+        // Safety: Since `state.have` is at least 1, `state.next` points to at least
+        // one readable byte within `state.output`.
+        let ret = unsafe { *state.next };
+        // Safety: Since `state.have` is at least 1, the byte between `state.next` and
+        // `state.next + 1` is within the bounds of the `state.output` buffer, as required
+        // by the pointer `add` method.
+        state.next = unsafe { state.next.add(1) };
+        return ret as _;
```

Comment:
```suggestion
        return c_int::from(ret);
```

using `from` is a signal that no truncation is happening. `From` for the builtin numeric types is only implemented when the cast is lossless.